src/apps/eth/eth_sign_msg.c

```diff
@@ -60,8 +60,13 @@ app_eth_sign_error_t app_eth_sign_msg(
     size_t payload_offset = snprintf(msg, sizeof(msg), "%s%d", msg_header, request->msg.size);
     memcpy(&msg[payload_offset], request->msg.bytes, request->msg.size);
 
-    if (!rust_workflow_verify_message(rust_util_bytes(request->msg.bytes, request->msg.size))) {
+    switch (rust_workflow_verify_message(rust_util_bytes(request->msg.bytes, request->msg.size))) {
+    case VerifyMessageResultInvalidInput:
+        return APP_ETH_SIGN_ERR_INVALID_INPUT;
+    case VerifyMessageResultUserAbort:
         return APP_ETH_SIGN_ERR_USER_ABORT;
+    default:
+        break;
```

Comment:
I would break on the specific `VerifyMessageResultOk` case and default to halt. This is in case we add another enum item and forget to handle it here.

Reply:
good idea, done